### PR TITLE
Replace pdfjs-dist build with es5 version

### PR DESCRIPTION
**Pull request by njleonzhang**
`react-pdf` supports es5 itself, and its dependency `pdfjs-dist` supports es5 through `dist/es5` folder. for entire supporting es5, `react-pdf` should require files in `pdfjs-dist/dist/es5`. 
if we don't fix this issue, all users who depend on `react-pdf` have to analyze code of  `react-pdf`, and find `pdfjs-dist` causes es5 incompatibility, then add `pdfjs-dist` babel transpiling config in their bundle tool(webpack) config. it don't make sense.

### CODE REVIEW

**Comment by wojtekmaj** [comment]
What is the issue we're trying to solve here? Does it crash on any of the supported browsers? Please raise an issue first so we can reproduce and discuss.

**Comment by njleonzhang** [comment]
> What is the issue we're trying to solve here? Does it crash on any of the supported browsers? Please raise an issue first so we can reproduce and discuss.

Sorry, I have not found the browser list react-pdf supports. According to the document, I can know IE is not supported in v5.

```
// https://unpkg.com/react-pdf@5.3.0/dist/umd/entry.js
...
var pdfjs = _interopRequireWildcard(require("pdfjs-dist"));
...
```
`react-pdf` import `pdfjs-dist` in build code, this cause projects which use `react-pdf` import `pdfjs-dist` indirectly. `react-pdf` is es5 compatible, but `pdfjs-dist` is not  by default.

This causet, it causes the projects which use `react-pdf` crash browsers that do not support es6.

**Review by mrbaguvix — COMMENTED** [review]
@njleonzhang Have you been able to create an issue and link it to this PR?

**Comment by njleonzhang** [comment]
@mrbaguvix why we need a issue, it's the process for pr merging?

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -0,0 +1,6336 @@
+lockfileVersion: 5.3
```
Comment (wojtekmaj) [inline pnpm-lock.yaml:1]: Please remove.

Comment (njleonzhang) [reply pnpm-lock.yaml]: done @wojtekmaj

**Review by wojtekmaj — APPROVED** [review]

**Comment by yordis** [comment]
Any idea when it could be release?

**Comment by njleonzhang** [comment]
any plan to merge and release?

**Comment by wojtekmaj** [comment]
As of pdfjs-dist 2.9.359 `es5` directory is gone. There is `legacy` directory though.

Not sure how to proceed, as we should really be using non-legacy code here. Perhaps a workaround based on webpack alias might be of any help.

**Comment by njleonzhang** [comment]
ORZ!! The breaking change is released with a minor version?

**Comment by wojtekmaj** [comment]
Lol, that's classic PDF.js. Their versions do not follow semver at all. That's the very reason we have `pdfjs-dist`(and only `pdfjs-dist`) version locked.

**Comment by chinnick967** [comment]
I'm having this issue https://github.com/wojtekmaj/react-pdf/issues/808 in our project and wanted to see when this PR will be merged and released. Thanks!

**Review by wojtekmaj — CHANGES_REQUESTED** [review]
Need to rebase off newest `main` and change paths to from /es5 to /legacy

**Comment by andredantasrocha** [comment]
@njleonzhang Thanks for the changes!!! can't wait to use the new code after the PR is merged :)

**Comment by wojtekmaj** [comment]
Thanks!

**Comment by andredantasrocha** [comment]
Amazing! Thanks guys!

**Pull request merged**